crates/project-model/src/project_json.rs

```diff
@@ -93,31 +94,83 @@ pub struct Crate {
     pub build: Option<Build>,
 }
 
-/// Additional metadata about a crate, used to configure runnables.
+/// Additional, build-specific data about a crate.
+///
+/// This is primarily used to
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct Build {
-    /// The name associated with this crate, according to the custom
-    /// build system being used.
+    /// The name associated with this crate.
+    ///
+    /// This is determined by the build system that produced
+    /// the `rust-project.json` in question. For instance, if buck were used,
+    /// the label might be something like `//ide/rust/rust-analyzer:rust-analyzer`.
+    ///
+    /// Do not attempt to parse the contents of this string; it is a build system-specific
+    /// identifier similar to [Crate::display_name].
     pub label: String,
     /// Path corresponding to the build system-specific file defining the crate.
+    ///
+    /// It is roughly analagous to [ManifestPath], but none of the indexing/loading
+    /// operations can be directly applied/used with this file.
     pub build_file: Utf8PathBuf,
-    /// What kind of target is this crate? For example, we don't want
-    /// to offer a 'run' button for library crates.
+    /// The kind of target.
+    ///
+    /// Examples (non-exhaustively) include [TargetKind::Lib], [TargetKind::Lib],
+    /// and [TargetKind::Test]. This information is used to determine what sort
+    /// runnable codelens to provide, if any.
     pub target_kind: TargetKind,
 }
 
+/// A template for runnables.
+///
+/// This to create runnables for running/debugging binaries and tests.
+/// In the future, it can be used to setup Flycheck.
+///
+/// # Example
+///
+/// Below is an example of a test runnable. `{label}` and `{test_id}`
+/// are explained in [Runnable::args]'s documentation.
+///
+/// ```json
+/// {
+///     "program": "buck",
+///     "args": [
+///         "test",
+///          "{label}",
+///          "--",
+///          "{test_id}",
+///          "--print-passing-details"
+///     ],
+///     "cwd": "/Users/dbarsky/fbsource",
```

Comment:
How about something more generically named here? Buck conventions aren't super useful.

Reply:
whoops, i thought i removed this! I'll make this something like `/home/user/repo-root/`.